kubernetes/utils/create_from_yaml.py

```diff
@@ -60,46 +66,69 @@ def create_from_yaml(
         processing of the request.
         Valid values are: - All: all dry run stages will be processed
     """
+    if path.exists(yaml_file):
+        with open(path.abspath(yaml_file)) as f:
+            content = f.read()
```

Comment:
what is your intention here?
why are you using StringIO instead of just calling `yml_document_all = yaml.safe_load_all(f)`?

Reply:
I think that was my comment from the old PR too.

Reply:
@juliantaylor, it feels like I am not getting through here. As I said before, I am using StringIO, so I can pass either a string containing a valid yaml or a file name.
This was desired by the reviewers of this PR in order to keep it backwards compatible. Originally I added 2 functions:
1. `create_from_file` which took over the functionality of `create_from_yaml`
2. `create_from_dict` 
Also, I modified `create_from_yaml` to read a string.
This was rejected in favor of having a single function with both functionality (I don't like this either, but again, the `create_from_yaml` is not reflecting correctly what the function does).

Reply:
it is not necessary for backward compatilbiity the old code only dealt with filenames (which is not a good api)

even if you need it, why the indirection over stringio? the yaml library can deal with file like inputs just fine

Reply:
most python libraries, including yaml loading from "files" accept file-like objects or data-strings as input.
the old `create_from_yaml` unfortunately uses file paths as input which is really poor as unnecessarily you tie the function to the filesystem and you cannot reasonably distinguish between something being a file path or data as they both just bytes.

Due to this original design changing this function to do something more useful is very ugly and I think it is not worthwhile.
Instead your new _from_dict function should be sufficient as it at least allows you to do more flexible things.

Reply:
> it is not necessary for backward compatilbiity the old code only dealt with filenames (which is not a good api)
Yes, the current API is not good and missleading.

> even if you need it, why the indirection over stringio? the yaml library can deal with file like inputs just fine.
This is not entirely correct, here how yaml loads a file name:
```
In [13]: !echo "{'foo': 'bar'}" > foo.txt
In [14]: yaml.safe_load('foo.txt')
Out[14]: 'foo.txt'
```
In any case, since you strongly object this, I will remove this.

Reply:
@juliantayloer, StringIO usage is now completely remove.

Reply:
yaml does not load filenames, it is better designed than the `create_from_yaml´
yaml loads file-like objects:

```
yaml.load(open("foo.txt"))